contracts/integrations/test/framework/actors/maker.ts

```diff
@@ -138,6 +139,92 @@ export function MakerMixin<TBase extends Constructor>(Base: TBase): TBase & Cons
+                [
+                    [this.actor, leftMakerToken],
+                    [this.actor, rightMakerToken],
+                    [this.actor, makerFeeToken],
+                    [taker, takerFeeToken],
+                ].map(async ([owner, token]) => {
+                    let balance = balanceStore.balances.erc20[owner.address][token.address];
+                    await (owner as Actor).configureERC20TokenAsync(token as DummyERC20TokenContract);
+                    balance = balanceStore.balances.erc20[owner.address][token.address] =
+                        constants.INITIAL_ERC20_BALANCE;
+                    return Pseudorandom.integer(0, balance.dividedToIntegerBy(8));
+                }),
+            );
+
+            // Select random amounts for the right order. The only constraint is that the slope
+            // of the left price curve is greater or equal to the inverse right price curve.
+            //   leftMakerAssetAmount/leftTakerAssetAmount >= rightTakerAssetAmount/rightMakerAssetAmount.
+            // We set the `rightMakerAssetAmount` equal to the `leftTakerAssetAmount` with probability 1/10,
+            // otherwise this scenario will never occur.
+            const shouldSetRightMakerEqualToLeftTakerAmount = Pseudorandom.integer(1, 10).eq(1);
+            const rightMakerAssetAmount = shouldSetRightMakerEqualToLeftTakerAmount
+                ? leftTakerAssetAmount
+                : Pseudorandom.integer(1, constants.INITIAL_ERC20_BALANCE.dividedToIntegerBy(8));
+            const rightTakerAssetAmount = Pseudorandom.integer(
+                1,
+                rightMakerAssetAmount.times(leftMakerAssetAmount).dividedToIntegerBy(leftTakerAssetAmount),
+            );
+
+            // Encode asset data
+            const [
+                leftMakerAssetData,
+                leftTakerAssetData,
+                rightMakerAssetData,
+                rightTakerAssetData,
+                makerFeeAssetData,
+                takerFeeAssetData,
+            ] = [leftMakerToken, leftTakerToken, rightMakerToken, rightTakerToken, makerFeeToken, takerFeeToken].map(
+                token =>
+                    this.actor.deployment.assetDataEncoder.ERC20Token(token.address).getABIEncodedTransactionData(),
+            );
+
+            // Construct and sign the left order
+            const leftOrder = await this.signOrderAsync({
+                makerAssetData: leftMakerAssetData,
+                takerAssetData: leftTakerAssetData,
+                makerFeeAssetData,
+                takerFeeAssetData,
+                makerAssetAmount: leftMakerAssetAmount,
+                takerAssetAmount: leftTakerAssetAmount,
+                makerFee,
+                takerFee,
+                feeRecipientAddress: Pseudorandom.sample(actors)!.address,
+            });
+
+            // Construct and sign the right order
+            const rightOrder = await this.signOrderAsync({
```

Comment:
So are the makers for these orders the same actor?

Reply:
Yeah, I figured this should be fine for fuzzing computations since the trade directions are validated by unit tests.

Reply:
Is it worth the effort to give them different makers? Seems a little strange that we're only fuzzing a fundamentally unlikely scenario.